catalog/app/containers/Admin/RolesAndPolicies/BucketsPermissions.tsx

```diff
@@ -0,0 +1,308 @@
+        <M.Typography variant="h6">Bucket access</M.Typography>
+        <M.Tooltip
+          arrow
+          title={
+            <>
+              Admin users can see all the buckets, but can only access the contents of the
+              buckets according to their assigned role (policies). Regular users can only
+              see the buckets configured in their assigned role (policies).
+            </>
+          }
+        >
+          <M.Icon fontSize="small" color="disabled" className={classes.icon}>
+            info_outlined
+          </M.Icon>
+        </M.Tooltip>
+        {bucketsData.case({
+          data: () => null,
+          fetching: () => (
+            <M.Tooltip arrow title="Fetching buckets">
+              <M.CircularProgress
+                size={20}
+                style={{ opacity: 0.3 }}
+                className={classes.icon}
+              />
+            </M.Tooltip>
+          ),
+          error: (e) => (
+            <M.Tooltip arrow title={<>Error fetching buckets: {e.message}</>}>
+              <M.Icon fontSize="small" color="disabled" className={classes.icon}>
+                error
+              </M.Icon>
+            </M.Tooltip>
+          ),
+        })}
+      </div>
+      {!!onAdvanced && (
+        <M.FormHelperText>
+          Manage access using per-bucket permissions or{' '}
+          <StyledLink onClick={onAdvanced}>set existing policy via ARN</StyledLink>
+        </M.FormHelperText>
+      )}
+      <M.Collapse in={!!error}>
+        <M.FormHelperText error>{error ? errors[error] || error : ' '}</M.FormHelperText>
+      </M.Collapse>
+
+      <M.List dense disablePadding>
+        {value.map((perm) => (
+          // XXX: navigate to bucket on click?
+          <M.ListItem
+            key={perm.bucket.name}
+            disableGutters
+            button
+            onClick={(event) => openPermissionMenu(event, perm)}
+          >
+            <M.ListItemAvatar style={{ minWidth: 44 }}>
+              <M.Avatar
+                style={{ width: 32, height: 32 }}
+                src={perm.bucket.iconUrl || defaultBucketIcon}
+              />
+            </M.ListItemAvatar>
+            <M.ListItemText
+              primary={
+                <>
+                  s3://{perm.bucket.name}{' '}
+                  <M.Box component="span" color="text.secondary" ml={0.5}>
+                    {perm.bucket.title}
+                  </M.Box>
+                </>
+              }
+            />
+            <M.ListItemSecondaryAction style={{ right: 0 }}>
+              <M.Tooltip
+                title={`Read-${perm.level === Level.READ ? 'only' : 'write'} access`}
+              >
+                <M.IconButton
+                  onClick={(event) => openPermissionMenu(event, perm)}
+                  edge="end"
+                  size="small"
+                >
+                  <M.Icon>{perm.level === Level.READ ? 'visibility' : 'edit'}</M.Icon>
```

Comment:
I spent 10 minutes debugging this. Realizing it's not a bug :)
It's confusing, that the edit button does the same as the visibility icon. I'd suggest splitting it into two elements:
1. show status. Words are better than icons. It's not obvious that the eye icon means read-only
2. button to trigger menu

Reply:
i'm not sure how to do this in MUI, bc ListItem only has secondary action which is an IconButton (well, i guess you can put smth like a select there, but it doesn't look natural)

Reply:
Probably, you can use table instead of a list

Reply:
well, table was the original layout, but list seems to suite this data better (apart from the icon/select problem), since there's not that many "fields" to represent this as a table (only bucket name and access level)